runtime/src/bank.rs

```diff
@@ -626,12 +627,73 @@ impl Bank {
             &sysvar::rewards::create_account(1, validator_point_value, storage_point_value),
         );
 
+        let validator_rewards = self.pay_validator_rewards(validator_point_value);
+
         self.capitalization.fetch_add(
-            (validator_rewards + storage_rewards) as u64,
+            validator_rewards + storage_rewards as u64,
             Ordering::Relaxed,
         );
     }
 
+    /// iterate over all stakes, redeem vote credits for each stake we can
+    ///   successfully load and parse, return total payout
+    fn pay_validator_rewards(&self, validator_point_value: f64) -> u64 {
+        let stake_history = self.stakes.read().unwrap().history().clone();
+        self.stake_delegations()
+            .iter()
+            .map(
+                |(
+                    stake_pubkey,
+                    Delegation {
```

Comment:
How about just calling this `delegation` and using `delegation.voter_pubkey` below?

Reply:
I thought the `vote_pubkey` form looked better, matched `stake_pubkey`.  `voter_pubkey` is an ancient name, and changing it perturbs cli, rpc

Reply:
I'm not a fan of this style, but I suppose it's just personal preference. I don't feel strongly that it needs to change.

Reply:
I changed it and prefer your style